### PR TITLE
Fix regressions and issues introduced in v0.51.0

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Improves null safety in generating reservations from allocations
- Reverts some optimizations to FRT calculation causing issues that prevent release to prod. These will be fixed later and reapplied then.
- Fixes an issues with unit reservation unit count counting archived reservation units
- Adds some missing fields to generated reservations from applications

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3437][TILA-3437]
- [TILA-3443][TILA-3443]

[TILA-3437]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3437
[TILA-3443]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3443

### CODE REVIEW

**Review by vergama — APPROVED** [review]
Nice

File: applications/tasks.py [hunk 1]
```diff
@@ -90,13 +119,13 @@ def generate_reservation_series_from_allocations(application_round_id: int) -> N
...
-                billing_address_street=application.billing_address.street_address,
-                billing_address_city=application.billing_address.city,
-                billing_address_zip=application.billing_address.post_code,
+                reservee_first_name=reservee_first_name,
+                reservee_last_name=reservee_last_name,
+                reservee_email=reservee_email,
+                reservee_phone=reservee_phone,
+                billing_address_street=billing_address_street,
+                billing_address_city=billing_address_city,
+                billing_address_zip=billing_address_zip,
```
[the remark this answers is not in the document]
Comment (matti-lamppu) [reply applications/tasks.py:127]: Added already in a [later commit](https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1251/commits/a675b8d5908d4ddc9414f3e9b88b1b1c72af7422).

**Pull request merged**